### PR TITLE
o/configcore: add flag that will determine if a system option can be

### PR DESCRIPTION
**Pull request by alfonsosanchezbeato**
used on a system with modes. Use it for the kernel command line
options.

### CODE REVIEW

File: overlord/configstate/configcore/runwithstate.go [hunk 1]
```diff
@@ -63,11 +63,11 @@ func init() {
 	addWithStateHandler(validateAutomaticSnapshotsExpiration, nil, validateOnly)
 
 	// netplan.*
-	addWithStateHandler(validateNetplanSettings, handleNetplanConfiguration, &flags{coreOnlyConfig: true})
+	addWithStateHandler(validateNetplanSettings, handleNetplanConfiguration, coreOnly)
 
 	// kernel.{,dangerous-}cmdline-append
 	// TODO we actually want this on classic with modes too
```
Comment (pedronis) [inline overlord/configstate/configcore/runwithstate.go:69]: the todo can be removed now?

File: overlord/configstate/configcore/kernel_test.go [hunk 2]
```diff
@@ -358,30 +363,43 @@ func (s *kernelSuite) testConfigureKernelCmdlineHappy(c *C, option []cmdlineOpti
 }
 
 func (s *kernelSuite) TestConfigureKernelCmdlineDangGrade(c *C) {
```
Comment (pedronis) [inline overlord/configstate/configcore/kernel_test.go:365]: not related to the PR, but I would expand all these Dang to Dangerous

File: overlord/configstate/configcore/kernel_test.go [hunk 3]
```diff
@@ -421,12 +441,13 @@ func (s *kernelSuite) TestConfigureKernelCmdlineConflict(c *C) {
...
-	err := configcore.Run(coreDev, rt)
+	err := configcore.Run(core20Dev, rt)
 	c.Assert(err, ErrorMatches, "kernel command line already being updated, no additional changes for it allowed meanwhile")
 }
 
 func (s *kernelSuite) TestConfigureKernelCmdlineSignedGradeNotAllowed(c *C) {
-	s.mockModel("signed")
+	isClassic := false
+	s.mockModelWithModeenv("signed", isClassic)
 	s.mockGadget(c)
```
Comment (pedronis) [inline overlord/configstate/configcore/kernel_test.go:451]: shouldn't this test use core20Dev too for consistency?

**Review by pedronis — COMMENTED** [review]
https://github.com/snapcore/snapd/pull/12605/commits/d2fee0415cfe71d120fdfbb0fa2a3f02b26e3394 looks good, some small comments

File: overlord/configstate/configcore/kernel_test.go (hunk 2 above)
Comment (alfonsosanchezbeato) [reply overlord/configstate/configcore/kernel_test.go]: Changed

File: overlord/configstate/configcore/kernel_test.go (hunk 3 above)
Comment (alfonsosanchezbeato) [reply overlord/configstate/configcore/kernel_test.go:451]: Good catch, changed

File: overlord/configstate/configcore/runwithstate.go (hunk 1 above)
Comment (alfonsosanchezbeato) [reply overlord/configstate/configcore/runwithstate.go]: Indeed, removed now

**Review by alfonsosanchezbeato — COMMENTED** [review]
Thanks, comments addressed now

File: tests/nested/manual/fde-on-classic/task.yaml [hunk 4]
```diff
@@ -111,6 +128,12 @@ execute: |
       remote.exec sudo grep -q -a '"This program cannot be run in XXX mode"' "$f"
   done
 
+  # Test appending values to the kernel command line
+  append_kernel_cmdline system.kernel.cmdline-append "append.val=1 append.flag other.val=foo"
```
Comment (mvo5) [inline tests/nested/manual/fde-on-classic/task.yaml:132]: This is very nice

**Review by mvo5 — APPROVED** [review]
Thank you!

File: overlord/configstate/configcore/kernel_test.go [hunk 5]
```diff
@@ -357,31 +362,44 @@ func (s *kernelSuite) testConfigureKernelCmdlineHappy(c *C, option []cmdlineOpti
...
 	s.testConfigureKernelCmdlineHappy(c,
 		[]cmdlineOption{{
 			name:    "system.kernel.cmdline-append",
 			cmdline: "par=val param star=val"}},
-		"signed")
+		"signed", isClassic)
 }
 
-func (s *kernelSuite) TestConfigureKernelCmdlineDangGradeDangCmdline(c *C) {
+func (s *kernelSuite) TestConfigureKernelCmdlineDangGradeDangerousCmdline(c *C) {
```
Comment (pedronis) [inline overlord/configstate/configcore/kernel_test.go:392]: nitpick: s/DangGrade/DangerousGrade/

**Review by pedronis — APPROVED** [review]
thank you, one nitpick

File: overlord/configstate/configcore/kernel_test.go (hunk 5 above)
Comment (alfonsosanchezbeato) [reply overlord/configstate/configcore/kernel_test.go]: Changed

**Comment by alfonsosanchezbeato** [comment]
Rebased now, also added an additional check in the spread test (change kernel command line, but instead of immediately rebooting, refresh the kernel).

**Pull request merged**